### ISSUE TITLE
8$: report E0429 for class 'extends' clause

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1114]
The following TypeScript code is invalid, but quick-lint-js accepts it without any diagnostic:
```typescript
export class D<T> {}
export class D2 extends D<<U>() => number> {}
```
quick-lint-js should report E0429 like it does for `implements` and `interface ... extends`.

**Comment by skdoosh** [comment]
@strager please close this issue as it has been solved by @arieldon in the pr in the conversation above.
